### ISSUE TITLE
Database connections are leaked while using the MySQL XA driver.

### ISSUE THREAD

**Issue by inf0rm4tik3r** [quarkusio/quarkus#11330]
**Describe the bug**
Connections to a MySQL 5.7 database were not closed properly.
At some point our Quarkus application could not properly handle a request. MySQL threw a "Connection limit reached" exception.

**Expected behavior**
Connections are not leaked.

**Actual behavior**
Connections were leaked (probably). At many points in time were more maximum connections open than defined through quarkus.datasource.jdbc.max-size.

**To Reproduce**
1. Configure the datasource to use the XA driver.
`com.mysql.cj.jdbc.MysqlXADataSource`

**To Solve**
Steps to solve the behavior:
1. Configure the datasource to use the standard JDBC driver.
`com.mysql.cj.jdbc.Driver`

**Configuration**
```properties
# Add your application.properties here, if applicable.

# Datasource
quarkus.datasource.url=${DATASOURCE_CONNECTION_URL:jdbc:mysql://localhost:3307/apprds?useUnicode=true&characterEncoding=UTF-8}
quarkus.datasource.transactions=enabled
quarkus.datasource.driver=com.mysql.cj.jdbc.Driver
quarkus.datasource.username=${DATASOURCE_USERNAME:root}
quarkus.datasource.password=${DATASOURCE_PASSWORD:admin}
quarkus.datasource.metrics.enabled=true
quarkus.hibernate-orm.dialect=org.hibernate.dialect.MySQL5Dialect
quarkus.hibernate-orm.database.charset=UTF-8
quarkus.hibernate-orm.log.sql=false

# Datasource connections and pooling
# -Enable datasource metrics collection.
# -If unspecified, collecting metrics will be enabled by default if the smallrye-metrics extension is active.
# -Default: unspecified
quarkus.datasource.jdbc.enable-metrics=true
# -The datasource pool minimum size, Default: 0
quarkus.datasource.jdbc.min-size=0
# -The datasource pool maximum size, Default: 20
quarkus.datasource.jdbc.max-size=20
# -The interval at which we validate idle connections in the background. Set to 0 to disable background validation.
# -Default: 2M
quarkus.datasource.jdbc.background-validation-interval=2M
# -The timeout before cancelling the acquisition of a new connection, Default: 5
quarkus.datasource.jdbc.acquisition-timeout=15S
# -The interval at which we check for connection leaks, Default: unspecified
quarkus.datasource.jdbc.leak-detection-interval=15M
# -The interval at which we try to remove idle connections, Default: 5M
quarkus.datasource.jdbc.idle-removal-interval=5M
# -The max lifetime of a connection, Default: unspecified
quarkus.datasource.jdbc.max-lifetime=15M
# -Disable pooling to prevent reuse of Connections.
# -Use this when an external pool manages the life-cycle of Connections.
# -Default: true
quarkus.datasource.jdbc.pooling-enabled=true

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)
<img width="690" alt="connections_correct" src="https://user-images.githubusercontent.com/9355833/89928137-a9ceb700-dc07-11ea-9baf-e103a240a8f3.png">
<img width="692" alt="connections_wrong" src="https://user-images.githubusercontent.com/9355833/89928140-aaffe400-dc07-11ea-8a82-c21af93f462c.png">

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`:  On all systems
 - Output of `java -version`:  11
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev:  1.6.1
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`):  3.6.3

**Additional context**
(Add any other context about the problem here.)

**Comment by famod** [comment]
Not really my field of expertise, but consulting the [datasources docs](https://quarkus.io/guides/all-config#quarkus-agroal_quarkus.datasource.jdbc.transactions) I noticed that you have set:
```
quarkus.datasource.transactions=enabled
```
instead of
```
quarkus.datasource.jdbc.transactions=xa
```
https://quarkus.io/guides/datasource#quarkus-agroal_quarkus.datasource.jdbc.transactions

`quarkus.datasource.transactions` (without the `jdbc` infix) doesn't seem to exist at all: https://quarkus.io/guides/all-config

/cc @Sanne

**Comment by inf0rm4tik3r** [comment]
@famod 
We used quarkus.datasource.transactions=xa before. But you are correct, it seems we were missing the jdbc part in the key. We did not yet really used XA transactions, so we could target the normal JDBC driver easily. We did not encounter any other problems other than the connections not being closed. Could this behaviour have arisen from the missing (correct) quarkus.datasource.jdbc.transactions=xa option?
